### PR TITLE
Add stack dumps from the daemon(s) on test timeout

### PR DESCRIPTION
**Pull request by cpuguy83**
Make test suites dump daemon stack on test timeout

Use `OnTimeout` (from revendor of go-check) callback on test timeouts to trigger a stack dump for
running daemons. This will help analyze stuck tests.

### CODE REVIEW

**Comment by cpuguy83** [comment]
Fitting that one of the experimental tests locked up and I got a stack dump from it :)

File: integration-cli/daemon.go [hunk 1]
```diff
@@ -268,6 +269,16 @@ func (d *Daemon) Kill() error {
...
 }
 
+// DumpStackAndQuit sends SIGQUIT to the daemon, which triggers it to dump its
+// stack to its log file and exit
+// This is used primarily for gathering debug information on test timeout
+func (d *Daemon) DumpStackAndQuit() {
+	if d.cmd == nil || d.cmd.Process == nil {
+		return
+	}
+	d.cmd.Process.Signal(os.Signal(syscall.SIGQUIT))
```
Comment (jhowardmsft) [inline integration-cli/daemon.go]: On Windows, this needs to signal the mutex instead. See https://github.com/jhowardmsft/docker-signal/blob/master/docker-signal.go for an example of how to do this.

Comment (LK4D4) [inline integration-cli/daemon.go]: ping @cpuguy83

Comment (cpuguy83) [inline integration-cli/daemon.go]: Fixed

**Comment by tonistiigi** [comment]
LGTM

**Comment by jhowardmsft** [comment]
LGTM

**Pull request merged**